src/java.base/share/classes/jdk/internal/vm/ThreadDumper.java

```diff
@@ -158,9 +160,9 @@ private static void dumpThreads(ThreadContainer container, PrintStream ps) {
 
     private static void dumpThread(Thread thread, PrintStream ps) {
         String suffix = thread.isVirtual() ? " virtual" : "";
-        ps.format("#%d \"%s\"%s%n", thread.threadId(), thread.getName(), suffix);
+        ps.println("#" + thread.threadId() + " \"" + thread.getName() + "\"" + suffix);
         for (StackTraceElement ste : thread.getStackTrace()) {
-            ps.format("      %s%n", ste);
+            ps.println("      " + ste);
```

Comment:
I'm curious if avoiding the string concat is more or less efficient for the volume of virtual threads and stack trace elements being written, though it does open the window to interleaved writes to the underlying stream.

```suggestion
            ps.print("      ");
            ps.println(ste);
```